tests/helpers/test_service.py

```diff
@@ -399,6 +411,28 @@ async def test_extract_entity_ids_from_area(hass, area_mock):
     )
 
 
+async def test_extract_entity_ids_from_devices(hass, area_mock):
```

Comment:
Can you add a test case where the entity is in another area than the device it's a part of?